test/powershell/Language/Scripting/ParameterBinding.Tests.ps1

```diff
@@ -0,0 +1,539 @@
+            $a
+        }
+
+        get-foo b | Should Be b
+    }
+
+    It 'Multiple positional parameters case 1' {
+       
+        function get-foo
+        {
+            param(
+            [Parameter(position=1)] $a,
+            [Parameter(position=2)] $b
+            )
+
+            $a
+            $b
+        }
+
+        (get-foo c d) -join ',' | Should Be 'c,d'
+
+        (get-foo -a c d) -join ',' | Should Be 'c,d'        
+
+        (get-foo -a c -b d) -join ',' | Should Be 'c,d'
+        
+        (get-foo -b d c) -join ',' | Should Be 'c,d'
+        
+        (get-foo c -b d) -join ',' | Should Be 'c,d'
+        
+    }
+
+    It 'Multiple positional parameters case 2:  the parameters are put in different order?' {
+        function get-foo
+        {
+            # the parameters are purposefully out of order.
+            param(
+            [Parameter(position=2)] $a,
+            [Parameter(position=1)] $b
+            )
+            $a
+            $b
+        }
+
+        (get-foo c d) -join ',' | Should Be 'd,c'        
+    }
+
+    It 'Value from pipeline' {
+        function get-foo
+        {
+            param(
+            [Parameter(valuefrompipeline=$true)] $a
+            )
+
+            process
+            {
+                if($a % 2 -eq 0)
+                {
+                    $a
+                }
+            }
+        }
+
+        (1..10 | get-foo) -join ',' | Should Be '2,4,6,8,10'
+    }
+
+    It 'Value from pipeline by property name' {
+        function get-foo
+        {
+            param(
+            [Parameter(valuefrompipelinebypropertyname=$true)] $foo
+            )
+
+            process
+            {
+                if($foo % 2 -eq 0)
+                {
+                    $foo
+                }
+            }
+        }
+
+        $b = 1..10 | select-object @{name='foo'; expression={$_ * 10}} | get-foo
+
+        $b -join ',' | Should Be '10,20,30,40,50,60,70,80,90,100'
+    }
+
+    It 'Value from remaining arguments' {
+        function get-foo
+        {
+            param(
+                [Parameter(position=1)] $a,
+                [Parameter(valuefromremainingarguments=$true)] $foo
+            )
+
+            $foo
+        }
+
+        ( get-foo a b c d ) -join ',' | Should Be 'b,c,d'
+
+        ( get-foo a b -a c d ) -join ',' | Should Be 'a,b,d'        
+
+        ( get-foo a b -a c -q d ) -join ',' | Should Be 'a,b,-q,d'
+    }
+
+    It 'Multiple parameter sets with Value from remaining arguments' {
+        function get-foo
+        {
+            param(
+                [Parameter(parametersetname='set1',position=1)] $a,
+                [Parameter(parametersetname='set2',position=1)] $b,
+                [parameter(valuefromremainingarguments=$true)] $foo
+            )
+
+            $foo
+        }       
+
+        { get-foo -a a -b b c d } | ShouldBeErrorId 'AmbiguousParameterSet,get-foo'
+        
+
+        ( get-foo -a a b c d ) -join ',' | Should Be 'b,c,d'
```

Comment:
I wonder if it's cleaner/better to add a filter to do the join, e.g.
```powershell
get-foo -a a b c d | Join ',' | Should Be 'b,c,d'
```
Pros - it feels more like PowerShell
Cons - it would be slower